server/routes/creatingReports/reportMayAlreadyExist.test.ts

```diff
@@ -0,0 +1,143 @@
+          expect(offenderService.getOffenderDetails).toBeCalledWith('user1-system-token', 2)
+        })
+    })
+
+    it('should get in-progress report', () => {
+      return request(app)
+        .get('/report/3/report-may-already-exist')
+        .expect('Content-Type', /html/)
+        .expect(() => {
+          expect(reportService.getReportInProgress).toBeCalledTimes(1)
+          expect(reportService.getReportInProgress).toBeCalledWith(3)
+        })
+    })
+
+    it('should redirect if no in-progress report', () => {
+      reportService.getReportInProgress.mockResolvedValue(null)
+      return request(app)
+        .get('/report/3/report-may-already-exist')
+        .expect(302)
+        .expect('Location', '/report/3/report-has-been-deleted')
+    })
+
+    it('should get reports that have same incident date as the new in-progress report', () => {
+      reportService.getReportInProgress.mockResolvedValue({ id: 15, incidentdate: '10/07/2021' })
+      return request(app)
+        .get('/report/3/report-may-already-exist')
+        .expect('Content-Type', /html/)
+        .expect(() => {
+          expect(reportService.getReportInProgress).toBeCalledTimes(1)
+          expect(reportService.getReportsByDate).toBeCalledWith(3, '10/07/2021')
+        })
+    })
+
+    it('should render page content', () => {
+      offenderService.getOffenderDetails.mockResolvedValue({ displayName: 'Erik Furtney' })
+      locationService.getLocation.mockResolvedValue({ userDescription: 'Location 1' })
+      reportService.getReportsByDate.mockResolvedValue([
+        {
+          date: moment('10/07/2021', 'DDMMYYYY'),
+          form: { incidentDetails: { locationId: 'room 1' } },
+          reporter: 'Prison Officer One',
+          status: 'SUBMITTED',
+        },
+        {
+          date: moment('10/07/2021', 'DDMMYYYY'),
+          form: { incidentDetails: { locationId: 'room 2' } },
+          reporter: 'Prison Officer Two',
+          status: 'SUBMITTED',
+        },
+      ])
+
+      return request(app)
+        .get('/report/1/report-may-already-exist')
+        .expect('Content-Type', /html/)
+        .expect(res => {
+          expect(res.text).toContain('A report for this incident may already exist')
+          expect(res.text).toContain('Erik Furtney')
+          expect(res.text).toContain('Saturday 10 Jul 2021, 00:00')
+          expect(res.text).toContain('Prison Officer One')
+          expect(res.text).toContain('Prison Officer Two')
+          expect(res.text).toContain('Location 1')
+        })
+    })
+
+    it('should render radios plus save and continue button', () => {
+      return request(app)
+        .get('/report/1/report-may-already-exist')
+        .expect('Content-Type', /html/)
+        .expect(res => {
+          expect(res.text).toContain(
+            '<input class="govuk-radios__input" id="cancel-report-yes" name="cancelReport" type="radio" value="true" data-qa="yes">'
```

Comment:
This style of testing is really coupled to the frontend - it doesn't really provide any benefit it's also covered by integration tests. 

There's no conditional logic to test here - not really worth keeping

Reply:
deleted the tests checking text content